sdk1/src/test/java/com/amazonaws/services/dynamodbv2/datamodeling/internal/AttributeValueMarshallerTest.java

```diff
@@ -266,9 +266,12 @@ public void testSimpleMapWithNull() {
       marshall(av);
       Assert.fail("Unexpected success");
     } catch (final NullPointerException npe) {
-      Assert.assertEquals(
-          "Encountered null map value for key NullKeyValue while marshalling attribute value {M: {KeyValue={S: ValueValue,}, NullKeyValue=null},}",
-          npe.getMessage());
+      // Map entries may permute under nondeterministic Java API
+      String npeMessage = npe.getMessage();
+      String common = "Encountered null map value for key NullKeyValue while marshalling attribute value ";
+      String case1 = common + "{M: {KeyValue={S: ValueValue,}, NullKeyValue=null},}";
+      String case2 = common + "{M: {NullKeyValue=null, KeyValue={S: ValueValue,}},}";
```

Comment:
Instead of hardcoding the two cases, I suggest we use TreeMap here as well. It's less brittle and easy to understand.

Reply:
I'll do it. Just to mention that to avoid hardcoding I need to make some changes to `AttributeValueMarshaller.java` , so there will be a bit changes in the source code as well!

Reply:
Ah, apologies, I see what you mean. Let's go back to trying to fix this test instead of making changes to `AttributeValueMarshaller.java` (We don't actually need to ensure ordering in the implementation itself). Your original approach lgtm :)